lib/abci/handlers/commitHandlerFactory.js

```diff
@@ -65,6 +104,65 @@ function commitHandlerFactory(
       blockExecutionContext.reset();
     }
 
+    rootTree.rebuild();
+
+    // Commit previous block data
+    if (previousBlockExecutionStoreTransactions) {
+      // Create document databases in previous dbs
+      const previousDataContractTransaction = previousBlockExecutionStoreTransactions.getTransaction('dataContracts');
+      const { updates: previousCreatedDataContracts } = previousDataContractTransaction.toObject();
+
+      const createDatabasePromises = Object.values(previousCreatedDataContracts)
+        .map(async (serializedDataContract) => {
+          const dataContract = await transactionalDpp.dataContract.createFromBuffer(
+            serializedDataContract,
+            {
+              skipValidation: true,
+            },
+          );
+
+          await previousDocumentDatabaseManager.create(dataContract);
+        });
+
+      await Promise.all(createDatabasePromises);
+
+      const previousDocumentsTransaction = previousBlockExecutionStoreTransactions.getTransaction('documents');
+
+      await populateMongoDbTransactionFromObject(
+        previousDocumentsTransaction.getMongoDbTransaction(),
+        previousDocumentsTransaction.toObject(),
+      );
+
+      // Commit previous block changes from the previous db transactions to previous databases
+      await previousBlockExecutionStoreTransactions.commit();
+    }
+
+    // Update and persist previous changes with changes from the current block
```

Comment:
```suggestion
    // Update previous transactions with changes from the current block
```

Let's move it under persisting.